compiler/haskell-ide-core/src/Development/IDE/State/FileStore.hs

```diff
@@ -6,44 +6,74 @@
 module Development.IDE.State.FileStore(
     getFileExists, getFileContents,
     setBufferModified,
-    fileStoreRules
+    fileStoreRules,
+    VFSHandle(..),
+    makeVFSHandle,
+    makeLSPVFSHandle,
     ) where
 
 
 
 import           StringBuffer
 import Development.IDE.Orphans()
 
+import Control.Concurrent.STM
```

Comment:
You're using STM like they are MVars. I'd prefer direct use of something like a `Var`.

Reply:
Fair enough, I generally default to STM but I don’t feel strongly about this so happy to change it.